### PR TITLE
RHCLOUD-29084 Support event types subscribed by default in connectors

### PR DESCRIPTION
**Pull request by gwenneg**
This is the last part of the "event types subscribed by default" implementation.

### CODE REVIEW

File: recipients-resolver/src/test/java/com/redhat/cloud/notifications/recipients/resolver/RecipientsResolverTest.java
Comment (gwenneg) [inline recipients-resolver/src/test/java/com/redhat/cloud/notifications/recipients/resolver/RecipientsResolverTest.java:1]: Good luck reviewing that file 😁

File: recipients-resolver/src/main/java/com/redhat/cloud/notifications/recipients/resolver/RecipientsResolver.java [hunk 1]
```diff
@@ -1,93 +1,89 @@
...
+import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
-import java.util.stream.Collectors;
-
 
+import static java.util.stream.Collectors.toSet;
 
 @ApplicationScoped
 public class RecipientsResolver {
```
Comment (gwenneg) [inline recipients-resolver/src/main/java/com/redhat/cloud/notifications/recipients/resolver/RecipientsResolver.java:16]: Review tip: don't compare the old and new codes. Reading the new code and comments should be clear enough.

File: recipients-resolver/src/test/java/com/redhat/cloud/notifications/recipients/resolver/RecipientsResolverTest.java
Comment (josejulio) [reply recipients-resolver/src/test/java/com/redhat/cloud/notifications/recipients/resolver/RecipientsResolverTest.java:1]: image

File: recipients-resolver/src/main/java/com/redhat/cloud/notifications/recipients/resolver/RecipientsResolver.java [hunk 2]
```diff
@@ -1,93 +1,92 @@
...
+        });
+
+        return users;
     }
 
+    private static boolean containsIgnoreCase(Set<String> usernames, String username) {
+        if (usernames == null || usernames.isEmpty()) {
+            return false;
+        } else {
+            return usernames.stream()
```
Comment (g-duval) [inline recipients-resolver/src/main/java/com/redhat/cloud/notifications/recipients/resolver/RecipientsResolver.java:86]: Couldn't we transform subscribers and unsubscribers only once at the beginning of calling method ?
Thanks to that, we may also initialise once an empty Set if subscribers or unsubscribers is null

File: recipients-resolver/src/main/java/com/redhat/cloud/notifications/recipients/resolver/RecipientsResolver.java [hunk 3]
```diff
@@ -1,93 +1,92 @@
...
 
-        return users;
-    }
+            // Subscriptions are only considered if the user preferences are NOT ignored.
+            if (!request.isIgnoreUserPreferences()) {
+                if (subscribedByDefault) {
+                    // When subscribedByDefault is true, we need to remove from the users anyone who unsubscribed.
+                    return containsIgnoreCase(unsubscribers, user.getUsername());
+                } else {
+                    // When subscribedByDefault is false, we need to remove from the users anyone who DIT NOT subscribe.
```
Comment (g-duval) [inline recipients-resolver/src/main/java/com/redhat/cloud/notifications/recipients/resolver/RecipientsResolver.java:69]:
```suggestion
                    // When subscribedByDefault is false, we need to keep only subscribed users, by removing from the users anyone who DID NOT subscribe.
```

**Review by g-duval — APPROVED** [review]
LGTM with a small suggestion ;)

**Review by MikelAlejoBR — APPROVED** [review]
LGTM!

File: recipients-resolver/src/main/java/com/redhat/cloud/notifications/recipients/resolver/RecipientsResolver.java (hunk 2 above)
Comment (gwenneg) [reply recipients-resolver/src/main/java/com/redhat/cloud/notifications/recipients/resolver/RecipientsResolver.java]: Good point. I pushed something.

**Review by MikelAlejoBR — APPROVED** [review]

**Pull request merged**